src/components/FilterEditorWindow/config/RequestFromCode.js

```diff
@@ -69,8 +77,23 @@ async function discoverFromCode(collectionName, data) {
     };
   }
 
+  // Then score these random points
+  const idsToInclude = randomResponse.data.result.points.map((point) => point.id);
+  const mustFilter = [{ has_id: idsToInclude }];
+  data.reqBody.filter = originalFilter || {};
+  data.reqBody.filter.must = mustFilter.concat(data.reqBody.filter?.must || []);
```

Comment:
Based on L83, filter should always be defined 
```suggestion
  data.reqBody.filter = originalFilter || {};
  data.reqBody.filter.must = mustFilter.concat(data.reqBody.filter.must || []);
```